packages/core/src/animation/Animator.ts

```diff
@@ -313,101 +309,98 @@ export class Animator extends Component {
-    srcPlayData.state && this._prepareSrcCrossData(crossOwnerCollection, srcPlayData, crossCurveMark, true);
+    srcPlayData.state && this._prepareSrcCrossData(crossOwnerCollection, srcPlayData, crossCurveMark, layerIndex, true);
     // Add dest cross curve data.
-    this._prepareDestCrossData(crossOwnerCollection, animatorLayerData.destPlayData, crossCurveMark, true);
+    this._prepareDestCrossData(crossOwnerCollection, animatorLayerData.destPlayData, crossCurveMark, layerIndex, true);
   }
 
   private _prepareFixedPoseCrossFading(animatorLayerData: AnimatorLayerData): void {
-    const crossOwnerCollection = this._crossOwnerCollection;
+    const { crossOwnerCollection, destPlayData, crossCurveMark, layerIndex } = animatorLayerData;
 
     // Save current cross curve data owner fixed pose.
     for (let i = crossOwnerCollection.length - 1; i >= 0; i--) {
       const item = crossOwnerCollection[i];
       item.saveFixedPoseValue();
       // Reset destCurveIndex When fixed pose crossFading again.
-      item.crossDestCurveIndex = -1;
+      item.crossDestCurveIndices[layerIndex] = -1;
     }
     // prepare dest AnimatorState cross data.
-    this._prepareDestCrossData(
-      crossOwnerCollection,
-      animatorLayerData.destPlayData,
-      animatorLayerData.crossCurveMark,
-      true
-    );
+    this._prepareDestCrossData(crossOwnerCollection, destPlayData, crossCurveMark, layerIndex, true);
   }
 
   private _prepareSrcCrossData(
-    crossCurveData: AnimationCurveOwner<KeyframeValueType>[],
+    crossOwnerCollection: AnimationCurveOwner<KeyframeValueType>[],
     srcPlayData: AnimatorStatePlayData,
     crossCurveMark: number,
+    layerIndex: number,
     saveFixed: boolean
   ): void {
     const { curveOwners } = srcPlayData.stateData;
     for (let i = curveOwners.length - 1; i >= 0; i--) {
       const owner = curveOwners[i];
       if (!owner) continue;
-      owner.crossCurveMark = crossCurveMark;
-      owner.crossCurveDataIndex = crossCurveData.length;
+      owner.crossCurveMarks[layerIndex] = crossCurveMark;
+      owner.crossOwnerCollectionIndices[layerIndex] = crossOwnerCollection.length;
       saveFixed && owner.saveFixedPoseValue();
-      this._addCrossCurveData(crossCurveData, owner, i, -1);
+      this._addCrossOwner(crossOwnerCollection, owner, layerIndex, i, -1);
     }
   }
 
   private _prepareDestCrossData(
-    crossCurveData: AnimationCurveOwner<KeyframeValueType>[],
+    crossOwnerCollection: AnimationCurveOwner<KeyframeValueType>[],
     destPlayData: AnimatorStatePlayData,
     crossCurveMark: number,
+    layerIndex: number,
     saveFixed: boolean
   ): void {
     const { curveOwners } = destPlayData.stateData;
     for (let i = curveOwners.length - 1; i >= 0; i--) {
       const owner = curveOwners[i];
       if (!owner) continue;
-      if (owner.crossCurveMark === crossCurveMark) {
-        crossCurveData[owner.crossCurveDataIndex].crossDestCurveIndex = i;
```

Comment:
```ts
owner.crossDestCurveIndex = i;
```
`rossCurveData[owner.crossCurveDataIndex]` and `owner` is the same instance

Reply:
resovled